kafka-client/src/main/java/dev/responsive/kafka/api/config/ResponsiveConfig.java

```diff
@@ -214,14 +214,26 @@ public class ResponsiveConfig extends AbstractConfig {
       + "will not be enabled. Setting this to a positive integer will enable async processing, but only if "
       + "there is at least one AsyncProcessor in the topology. See javadocs for AsyncProcessorSupplier for details.";
 
-  public static final String ASYNC_MAX_EVENTS_PER_KEY_CONFIG = "responsive.async.max.events.per.key";
-  private static final int ASYNC_MAX_EVENTS_PER_KEY_DEFAULT = 5;
-  private static final String ASYNC_MAX_EVENTS_PER_KEY_DOC = "The maximum number of pending events at a time for "
-      + "a given key. This puts a limit on how many events for each key can be either currently processing or awaiting "
-      + "processing by the async thread pool. Since all events with the same key must be processed in offset order, this"
+  public static final String ASYNC_MAX_EVENTS_QUEUED_PER_ASYNC_THREAD_CONFIG = "responsive.async.max.events.queued.per.async.thread";
+  private static final int ASYNC_MAX_EVENTS_QUEUED_PER_ASYNC_THREAD_DEFAULT = 5;
```

Comment:
@rodesai  As I said earlier, 2 felt really low to me, but I'm open to changing the default. For this and the per-key one

Reply:
These defaults look fine to me. I don't really see the benefit of accumulating lots of records in the queues. You're going to have to pay the cost at flush time anyway. One potential improvement would be to compute the default for `responsive.async.max.events.queued.per.async.thread` from the size of the thread pool (it could just be the size of the thread pool times some factor (even x1 seems fine - you have one event enqueued for each thread and one being actively processed))